### PR TITLE
allow insertion of UserActions that lack a title

### PR DESCRIPTION
**Pull request by dwrensha**
The `Action.title` field is deprecated, so we shouldn't require that it be present.

See also #2177.

### CODE REVIEW

**Comment by zarvox** [comment]
Do we want to require that at least one of (title, nounPhrase) is present?  Can that be expressed?

**Comment by dwrensha** [comment]
I think having neither is fine because we fall back to "grain".

**Comment by zarvox** [comment]
:shipit:

**Comment by kentonv** [comment]
Maybe we should go ahead and introduce a method that takes a package ID as an argument? It feels kind of scary to be round-tripping this structure through the client, even though I don't think it's exploitable (currently).

**Comment by kentonv** [comment]
Also I believe this is the only `allow()` rule we have in the whole codebase. Would be nice to have none.

**Comment by dwrensha** [comment]
Garply, retest this please.

**Comment by dwrensha** [comment]
The [transient failure on Jenkins](https://build.sandstorm.io/job/Sandstorm%20PRs%20-%20build/668/) seems to indicate some kind of race condition. I was unable to reproduce it locally. I'm not sure whether or not it is related to my changes in this pull request.

File: shell/packages/sandstorm-db/db.js [hunk 1]
```diff
@@ -1257,42 +1257,6 @@ _.extend(SandstormDb.prototype, {
     return value;
   },
 
-  addUserActions: function (packageId) {
```
Comment (kentonv) [inline shell/packages/sandstorm-db/db.js]: This method is still in-use for auto-updates, which now seem to be broken.

**Pull request merged**